cmplibrary/src/main/java/com/sourcepoint/cmplibrary/core/web/ConsentWebView.kt

```diff
@@ -107,9 +106,7 @@ internal class ConsentWebView(
         consent: String?,
     ): Either<Boolean> = check {
 
-        if (connectionManager.isConnected.not()) throw NoInternetConnectionException(
```

Comment:
We need the ConnectionManager, the `connectionManager.isConnected` was good

Reply:
even though it was ok to use `connectionManager.isConnected` previously, the `ConnectionManager` itself did not took into account one deprecated feature (`NetworkInfo` class is deprecated since `API 29`)

when I started adding a check for this deprecated feature, I thought that we do not need a separate object to check Internet connection, since:
1. the only purpose of the `ConnectionManager` is to check Internet connection, which is ok, but nothing more, no checks for a specific connection type or whatever, so there is no need in creating a specific class for that
2. the `ConnectionManager` object uses context instance, for which we can actually create an extension function and that would do the trick

overall, I thought that creating `ConnectionManager` is a bit of overengineering, so that is why I decided to entirely remove `ConnectionManager` and use `Context.isInternetConnected()` extension function

Reply:
The ConnectionManager follows  the Single responsibility principle which we use to isolate the Context. 
The context cannot be replace with a mock object.
Each feature that we add should have a UI test and this is the easy and controllable way of doing it